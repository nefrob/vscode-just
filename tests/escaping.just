

# Test braces escape content

@b1 var="1":
  # 1
  echo {{var}}
  echo {{ var }}
  echo '{{var}}'
  echo '{{ var }}'
  echo '''{{var}}'''
  echo '''{{ var }}'''

  # var
  echo {{'var'}}
  echo {{ 'var' }}

  # strange
  echo '{{'var'}}'
  echo '''{{'''var'''}}'''
  echo '{{' var '}}'
  echo '''{{''' var '''}}'''
  echo '{{ 'var' }}'
  echo '''{{ '''var''' }}'''
  echo ' {{ 'var' }} '
  echo ''' {{ '''var''' }} '''
  echo '{{ 'a' + var +  'b' }}'

# Test braces are escaped

@b2 var="1":
  echo '{{{{ var }}'
  echo '''{{{{ var }}'''
  echo {{{{ var }}
  echo '{{ "{{" }} var }}'
  echo '''{{ "{{" }} var }}'''
  echo '''{{ """{{""" }} var }}'''
  echo {{ "{{" }} var }}
  echo {{ """{{""" }} var }}
  echo '{{'{{ var }}'}}'
  echo '''{{'''{{ var }}'''}}'''
  echo {{ '{{ var }}' }}
  echo {{ '''{{ var }}''' }}
  echo {{ "{{ var" }} }}
  echo {{ "{{ var }}" }}
  echo {{ "{{ 'var' }}" }}

# Illegal for justfiles

# echo '{{' {{ 'var' }} '}}'
# echo {{' {{ 'var' }} '}}
# echo {{{ var }}

  