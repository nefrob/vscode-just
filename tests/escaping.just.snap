># Escaped blocks
#^^^^^^^^^^^^^^^^ source.just comment.line.number-sign.just
>
>publish:
#^^^^^^^ source.just entity.name.function.just
#       ^ source.just keyword.operator.recipe.end.just
>  rm -f {{tarball}}
#^^^^^^^^ source.just
#        ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#          ^^^^^^^ source.just string.interpolated.escaping.just
#                 ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
>  mkdir {{tardir}}
#^^^^^^^^ source.just
#        ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#          ^^^^^^ source.just string.interpolated.escaping.just
#                ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
>  cp README.md *.c {{tardir}}
#^^^^^^^^^^^^^^^^^^^ source.just
#                   ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#                     ^^^^^^ source.just string.interpolated.escaping.just
#                           ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
>  tar zcvf {{tarball}} {{tardir}}
#^^^^^^^^^^^ source.just
#           ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#             ^^^^^^^ source.just string.interpolated.escaping.just
#                    ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#                      ^ source.just
#                       ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#                         ^^^^^^ source.just string.interpolated.escaping.just
#                               ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
>  scp {{tarball}} me@server.com:release/
#^^^^^^ source.just
#      ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#        ^^^^^^^ source.just string.interpolated.escaping.just
#               ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#                 ^^^^^^^^^^^^^^^^^^^^^^ source.just
#                                       ^ source.just keyword.operator.path-join.just
>  rm -rf {{tarball}} {{tardir}}
#^^^^^^^^^ source.just
#         ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#           ^^^^^^^ source.just string.interpolated.escaping.just
#                  ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#                    ^ source.just
#                     ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#                       ^^^^^^ source.just string.interpolated.escaping.just
#                             ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
>
>braces:
#^^^^^^ source.just entity.name.function.just
#      ^ source.just keyword.operator.recipe.end.just
>  echo 'I {{{{LOVE}} curly braces!'
#^^^^^^^ source.just
#       ^ source.just string.quoted.single.just
#        ^^^^^^^^^^^^^^^^^^^^^^^^^^ source.just string.quoted.single.just
#                                  ^ source.just string.quoted.single.just
>
>braces:
#^^^^^^ source.just entity.name.function.just
#      ^ source.just keyword.operator.recipe.end.just
>  echo '{{'I {{LOVE}} curly braces!'}}'
#^^^^^^^ source.just
#       ^^^^ source.just string.quoted.double.indented.just
#           ^^ source.just
#             ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#               ^^^^ source.just string.interpolated.escaping.just
#                   ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#                     ^^^^^^^^^^^^^^ source.just
#                                   ^ source.just string.quoted.single.just
#                                    ^^ source.just string.quoted.single.just
#                                      ^ source.just string.quoted.single.just
>
>braces:
#^^^^^^ source.just entity.name.function.just
#      ^ source.just keyword.operator.recipe.end.just
>  echo 'I {{ "{{" }}LOVE}} curly braces!'
#^^^^^^^ source.just
#       ^ source.just string.quoted.single.just
#        ^^ source.just string.quoted.single.just
#          ^^ source.just string.quoted.single.just string.interpolated.escaping.just string.interpolated.escape.just
#            ^ source.just string.quoted.single.just string.interpolated.escaping.just
#             ^^^^ source.just string.quoted.single.just string.interpolated.escaping.just string.quoted.double.indented.just
#                 ^ source.just string.quoted.single.just string.interpolated.escaping.just
#                  ^^ source.just string.quoted.single.just string.interpolated.escaping.just string.interpolated.escape.just
#                    ^^^^^^^^^^^^^^^^^^^^ source.just string.quoted.single.just
#                                        ^ source.just string.quoted.single.just